src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs

```diff
@@ -192,7 +217,7 @@ public new CSharpCommandLineArguments Parse(IEnumerable<string> args, string? ba
                         continue;
 
                     case "langversion":
```

Comment:
Mostly I was going for option names that were likely to appear multiple times in standard compilation requests. For example `/reference` was an easy win because there are hundreds of those per compilation. The next one after that was `/analyzer`. After that though pretty much every other option is used only once. Wasn't making big impact. 

Don't have any objections to pushing more up to the fast path though.

Reply:
That makes sense and seems like a reasonable cutoff.